lambdas/service/app.py

```diff
@@ -623,7 +623,9 @@ def validate_catalog(catalog):
     except RequirementError as e:
         raise BadRequestError(e)
     if catalog not in config.catalogs:
-        raise BadRequestError(f'Catalog name {catalog!r} is invalid must be within {[*config.catalogs]!r}')
+        raise BadRequestError(f'Catalog name {catalog!r} is invalid. Must be one of {set([*config.catalogs])}.')
```

Comment:
This is quirky. `set()` accepts an iterable directly.

Reply:
there was an issue with this due to `config.catalogs` being a mapping.
```
set(*config.catalogs) -> {'T', 'E', 'S'}   a set for the string 'TEST'
set([*config.catalogs]) -> {'test'}
set(config.catalogs.keys()) -> {'test'}
```
Further i think this will also result in a internal server error when its run on the lambda, off the testing env.
```
>from azul import config
>config.catalogs
{'dcp1': {'repository': 'dss', 'metadata': 'hca'}, 'it1': {'repository': 'dss', 'metadata': 'hca'}}
>a = set([*config.catalogs])
>a
{'dcp1', 'it1'}
>set(*config.catalogs)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: set expected at most 1 argument, got 2
```

I'd be open to both, if you don't like the quirkiness of [*config.catalogs], we can use config.catalogs.keys().
Either way its temporary until #1465 is merged, at which point this is all removed.

Reply:
```
>>> from azul import config
>>> set(config.catalogs)
{'dcp2ebi', 'dcp1', 'it2', 'it1', 'it2ebi', 'dcp2'}
```